### DISCUSSION TITLE
Tooltip on Pixi Container

### DISCUSSION THREAD

**Discussion by 01chirag** [pixijs/pixijs#7796] [Q&A]
Hey Guys,
I am new to Pixijs and facing one issue while using it in reactjs application. I am not sure if this is a bug or I am doing something incorrectly.
Let me explain what I am trying to do.
I want to show tooltip on the sprites which I have in Pixi container.
Tooltip is visible however some of the part is shown behind the other sprites to fix that I tried to use zindex but it is not working somehow. Also tried with the [package](https://github.com/pixijs/layers), but somehow it is also not working. 
These are the screens how it is visible
![screenshot-newtab-2021 09 14-19_55_56](https://user-images.githubusercontent.com/90705313/133278193-120520bb-0c86-4286-a0a8-e99bf06415bf.png)
![screenshot-newtab-2021 09 14-19_54_57](https://user-images.githubusercontent.com/90705313/133278201-39d419a6-4d29-4301-94ec-972c70de00cb.png)

This is the code snippet

```
import React from "react"
import * as PIXI from "pixi.js"
import ObjectStorageSprite from "../../../src/assets/sprites/object_storage.png"
import MessageBubble from "../../../src/assets/sprites/message.png"
  var  app = new PIXI.Application({width: window.innerWidth , height: window.innerHeight,backgroundColor: 0xFFFFFF,});
  var  loader = new PIXI.Loader();
  var  pixi_cnt = null;
  var  resources = null;
  
  function MapContainer(props)  {

  const updatePixiCnt= (element) => {
    pixi_cnt = element;
    if(pixi_cnt) {
       pixi_cnt.appendChild(app.view);
       loader
        .add("object_storage", ObjectStorageSprite)
        .load((loader, tempResources) => {
          resources = tempResources;
          initialize();
        });
    }
  };


  const initialize = () => {
    let x = 10 ;
    let y =10;
    let i =0;
    var spContainer = new PIXI.Container();
    spContainer.sortChildren = true;
    spContainer.sortDirty = true;
    while(i< 5) {
      i +=1;
      let image = "object_storage";
      let sprite = new PIXI.Sprite(resources[image].texture);
      sprite.x = x;
      sprite.y = y;
      const objectStorageMouseOverFun = (e) => {
        mouseOverFun("dummy text",sprite,0);
      }
      const objectStorageMouseOutFun = (e) => {
        mouseOutFun(sprite);
      }
      sprite.interactive = true;
      sprite.buttonMode = true;
      sprite.on("mouseover",objectStorageMouseOverFun);
      sprite.on("mouseout",objectStorageMouseOutFun);
      sprite.zIndex = 10;
      sprite.width = 150;
      sprite.height = 150;
      x += 100;
      spContainer.addChild(sprite);
    }
    app.stage.addChild(spContainer);

  }
  const mouseOverFun = (message,sprite,rotation) => {
    var toolTipText = new PIXI.Text(message,
                         {
                             fontFamily : 'Arial',
                             fontSize: 24,
                         });
    toolTipText.position.set(10,20);
    var toolTip=PIXI.Sprite.from(MessageBubble);
    toolTip.addChild(toolTipText);
    toolTip.rotation = -rotation ;
    toolTip.zIndex = 11;                      
    sprite.message = toolTip;
    sprite.addChild(toolTip);               
  }

  const mouseOutFun = (sprite) =>{
    sprite.removeChild(sprite.message);
    delete sprite.message;
  }
    return <div ref={updatePixiCnt}
          id="pixiMapContainer"
          style={{position: "relative"}}
          >
          </div>
          ;
}

export default MapContainer

```
### Environment

"pixi.js": "^6.1.2",
"react": "16.9.0"
Chrome: "Version 92.0.4515.131 (Official Build) (64-bit)"
Os Name : "Ubuntu 20.04.2 LTS"

**Comment by ivanpopelyshev** [comment]
Show me how it didnt work with layers. Better if you make codesandbox or jsfiddle out of it. 

HEre's like to `layers` for you, in case you use jsfiddle or codepen: https://cdn.jsdelivr.net/npm/@pixi/layers@latest/dist/pixi-layers.umd.js

**Reply by 01chirag** [reply]
I just got confused about the parent-child relationship after spending some time it worked without layers. Here is a working example
[Link](https://codesandbox.io/s/unruffled-butterfly-oj4qk?file=/src/App.js). This might help someone.

**Reply by ivanpopelyshev** [reply]
parent-child relation exists for childs to
1. add/remove when parent is added/removed
2. transform together
3. rendered together

Layers exists to remove condition "3",  and specify a "layer" where tooltop is rendered. if you add layer after everything to stage - it'll be shown on top.

That way you can omit this thing:
```
toolTip.x = sprite.position._x;
toolTip.y = sprite.position._y;
```

When you'll have many cases like this one - use layers. Otherwise, do whatever works for you :)
